### PR TITLE
API-1456: event subscription hit the timeout limit

### PR DESCRIPTION
**Pull request by tomglvng**

### CODE REVIEW

**Review by pchasle — COMMENTED** [review]
You are missing a unit test (phpspec) on `GuzzleWebhookClient.php` to test that the timeout error is logged.

File: src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventsApiDebugWebhookClientLogger.php [hunk 1]
```diff
@@ -10,14 +10,27 @@
  * @copyright 2020 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
  */
-interface EventsApiDebugResponseErrorLogger
+interface EventsApiDebugWebhookClientLogger
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventsApiDebugWebhookClientLogger.php:13]:
```suggestion
interface EventsApiRequestLogger
```
or something like that

File: src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventsApiDebugWebhookClientLogger.php [hunk 2]
```diff
@@ -10,14 +10,27 @@
...
+        array $events,
+        string $url,
+        int $statusCode,
+        array $headers
+    ): void;
+
+    /**
+     * @param array<WebhookEvent> $events
+     */
+    public function logTimeoutLimit(
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventsApiDebugWebhookClientLogger.php:30]:
```suggestion
    public function logEventsApiRequestTimedOut(
```

Comment (tomglvng) [reply src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventsApiDebugWebhookClientLogger.php:30]: It's ok for me ;)

**Review by pyaillet — APPROVED** [review]

**Review by pchasle — APPROVED** [review]

**Pull request merged**